src/assets/Generator.Shared/HttpPipelineExtensions.cs

```diff
@@ -57,43 +56,31 @@ public static Response ProcessMessage(this HttpPipeline pipeline, HttpMessage me
 
         public static async ValueTask<Response<bool>> ProcessHeadAsBoolMessageAsync(this HttpPipeline pipeline, HttpMessage message, ClientDiagnostics clientDiagnostics, RequestContext? requestContext)
         {
-            var (cancellationToken, statusOption) = ApplyRequestContext(requestContext);
-            await pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
-            switch (message.Response.Status)
+            var response = await pipeline.ProcessMessageAsync(message, clientDiagnostics, requestContext).ConfigureAwait(false);
+            switch (response.Status)
             {
                 case >= 200 and < 300:
-                    return Response.FromValue(true, message.Response);
+                    return Response.FromValue(true, response);
                 case >= 400 and < 500:
-                    return Response.FromValue(false, message.Response);
+                    return Response.FromValue(false, response);
                 default:
-                    var exception = await clientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
-                    if (statusOption == ErrorOptions.NoThrow)
-                    {
-                        return new ErrorResponse<bool>(message.Response, exception);
-                    }
-
-                    throw exception;
+                    var exception = await clientDiagnostics.CreateRequestFailedExceptionAsync(response).ConfigureAwait(false);
+                    return new ErrorResponse<bool>(response, exception);
```

Comment:
Is it correct to remove the check for status option here?  Would ProcessMessageAsync have handled any errors that come up?

Reply:
Yes, `ProcessMessageAsync` would throw already if `requestContext.ErrorOptions != ErrorOptions.NoThrow`